src/pages/signin/SignInPage.js

```diff
@@ -59,23 +74,37 @@ const defaultProps = {
 /**
  * @param {Boolean} hasLogin
  * @param {Boolean} hasValidateCode
+ * @param {Boolean} hasAccount
  * @param {Boolean} isPrimaryLogin
  * @param {Boolean} isAccountValidated
  * @param {Boolean} hasEmailDeliveryFailure
  * @returns {Object}
  */
-function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, hasEmailDeliveryFailure}) {
+function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, isSAMLEnabled, isSAMLRequired, isUsingSAMLLogin, hasEmailDeliveryFailure}) {
     const shouldShowLoginForm = !hasLogin && !hasValidateCode;
-    const shouldShowEmailDeliveryFailurePage = hasLogin && hasEmailDeliveryFailure;
+    let shouldShowSAMLEnabledForm = false;
+    let shouldInitiateSAMLLogin = false;
+    const platform = getPlatform();
+
+    // SAML is temporarily restricted to users on the beta or to users signing in on web and mweb
+    if (Permissions.canUseSAML() || platform === CONST.PLATFORM.WEB) {
+        shouldShowSAMLEnabledForm = hasAccount && hasLogin && isSAMLEnabled && !isSAMLRequired && isUsingSAMLLogin;
+        shouldInitiateSAMLLogin = hasAccount && hasLogin && isSAMLRequired;
```

Comment:
NAB, should we add `&& isSAMLEnabled`?

Reply:
No, because SAML if SAML is required it's also enabled 😅 the latter is the stricter condition of the two and we only want to initiate the SAML login process if they have it required (not if it's only enabled)

Reply:
Makes sense, just a safe check that it is enabled and required